content/articles/2017-12-14-styler-1-0-0.Rmd

```diff
@@ -0,0 +1,255 @@
+)
+```
+
+
+In addition, the `strict` option also adds braces (if `scope = "tokens"`) to conditional statements if they are multi-line:
+```{styler, strict = TRUE}
+if (set_q(x))
+    3 else 
+    5
+```
+
+
+### math token spacing
+
+styler can identify and handle unary operators and other math tokens:
+
+```{styler}
+1++1-1-1/2
+```
+
+This is tidyverse style. However, styler offers very granular control for math token spacing. Assuming you like spacing around `+` and `-`, but not around `/` and `*` and `^`. This can be achieved as follows:
+```{styler, math_token_spacing = specify_math_token_spacing(zero = c("'/'", "'*'", "'^'"))}
+1++1/2*2^2 
+```
+
+### reindention
+
+You can set the indention of lines that match a certain regular expression manually. For example, if you don't want comments starting with `###` to be indented, you can formulate an unindention rule:
+```{styler, reindention = specify_reindention(regex_pattern = "^###", indention = 0)}
+a <- function() {
+    ### not to be indented
+    # indent normally
+    33
+    }
+```
+
+
+These are options that allow flexible reformatting. You can even go further, as described in the ["Customizing styler"](http://styler.r-lib.org/articles/customizing_styler.html) vignette.
+
+# A few more examples
+
+Now that you know that the rules can be applied in a flexible fashion, let's focus on `scope = "tokens"` and `strict = TRUE` and illustrate how specific rules of the tidyverse style guide are implemented. 
+
+Spaces are placed around operators such as `=`, after (but not before) commas, before comments etc.
+```{styler}
+split_fast<-split_into_fractions(target="fast",origin  ="E2-k3",...)#comment
+```
+
+Function declarations are indented if multi-line:
+```{styler}
+my_fun <- function(x, 
+y, 
+z) {
+  just(z)
+}
+```
+
+styler can also format complicated expressions that involve line breaking and indention based on both brace expressions and operators:
+
+```{styler}
+if (x >3) {stop("this is an error")} else {
+c(there_are_fairly_long,
+1 / 33 * 
+2 * long_long_variable_names)%>% k(
+
+) }
+```
+
+Lines are broken after `(` if a function call spans multiple lines:
+
+```{styler}
+do_a_long_and_complicated_fun_cal("which", has, way, to, 
+                              "and longer then lorem ipsum in its full length"
+                              )
+```
+
+styler replaces `=` with `<-` for assignment if necessary:
+```{styler}
+one = 3
+data_frame(a = 3)
+```
+
+It also handles single quotes according to the tidyverse style guide:
+```{styler}
+'one string'
+'one string in a "string"'
+"one string in a 'string'"
+
+```
+
+Braces are added to function calls in pipes:
+
+```{styler}
+a %>%
+  b %>%
+  c
+```
+
+tidyeval syntax is handled properly:
+```{styler}
+mtcars %>%
+  group_by(!!!my_vars)
+```
+
+That was a quick overview of styler. Please feel free to grab styler from CRAN, run it on your own package and tell us what you think.
+
+
+```{styler, scope = "spaces"}
```

Comment:
Not sure how this chunk crept in but it should be removed. Gives the article a weird "tail" after the apparent conclusion.

Reply:
uggh right, thanks.